tests/build-tests.js

```diff
@@ -1,174 +1,203 @@
+                            {
+                                rule: 'f3.com/foo',
+                                domains: [
+                                    'dom1.com',
+                                    'dom0.com', // should be sorted in output
+                                    'dom1.com' // duplicate should be removed
+                                ],
+                                reason: 'Reason1.'
+                            }
+                        ]
+                    }
+                }
             }
-        ]
+        }
     }
 }
 
 // combined expected result
 const ta1plus2 = {
-    'f1.com': {
-        rules: [
-            {
-                rule: 'www.f1.com/foo/bar',
-                domains: [
-                    'dom0.com',
-                    'dom1.com',
-                    'dom2.net',
-                    'dom3.net'
-                ],
-                reason: 'Reason1.; Reason2.'
-            },
-            {
-                rule: 'www.f1.com/bar',
-                domains: [
-                    'dom1.com'
-                ],
-                reason: 'Reason1.'
-            },
-            {
-                rule: 'www.f1.com/foo',
-                domains: [
-                    'dom0.com',
-                    'dom1.com'
-                ],
-                reason: 'Reason1.'
-            }
-        ]
-    },
-    'f2.com': {
-        rules: [
-            {
-                rule: 'f2.com/foo',
-                domains: [
-                    'dom0.com',
-                    'dom1.com'
-                ],
-                reason: 'Reason1.'
-            }
-        ]
-    },
-    'f3.com': {
-        rules: [
-            {
-                rule: 'f3.com/foo',
-                domains: [
-                    'dom0.com',
-                    'dom1.com'
-                ],
-                reason: 'Reason1.'
-            }
-        ]
-    },
-    'f4.com': {
-        rules: [
-            {
-                rule: 'f4.com/foo.bar',
-                domains: [
-                    'dom1.com'
-                ],
-                reason: 'Reason1.'
-            },
-            {
-                rule: 'f4.com/foo',
-                domains: [
-                    'dom1.com'
-                ],
-                reason: 'Reason1.'
+    features: {
+        trackerAllowlist: {
+            state: 'enabled',
+            exceptions: [],
+            settings: {
+                allowlistedTrackers: {
+                    'f1.com': {
+                        rules: [
+                            {
+                                rule: 'www.f1.com/foo/bar',
+                                domains: [
+                                    'dom0.com',
+                                    'dom1.com',
+                                    'dom2.net',
+                                    'dom3.net'
+                                ],
+                                reason: 'Reason1.; Reason2.'
```

Comment:
hm, interesting, where is the code that does this concatenation of reasons? `arrayMerge` doesn't seem to be doing this?

Reply:
Hm, I see some more tests in the config-tests file. Why those are separate?